go/tools/go_bin_runner/main.go

```diff
@@ -34,8 +42,35 @@ func main() {
 		log.Fatal(err)
 	}
 
+	hashesBefore, err := hashWorkspaceRelativeFiles(cfg.DepsFiles)
+	if err != nil {
+		log.Fatal(err)
+	}
+
 	args := append([]string{goBin}, os.Args[1:]...)
-	log.Fatal(runProcess(args, env, os.Getenv("BUILD_WORKING_DIRECTORY")))
+	cwd := os.Getenv("BUILD_WORKING_DIRECTORY")
+	err = runProcess(args, env, cwd)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	hashesAfter, err := hashWorkspaceRelativeFiles(cfg.DepsFiles)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	diff := diffMaps(hashesBefore, hashesAfter)
+	if len(diff) > 0 {
+		if HasBazelModTidy == "True" {
+			_, _ = fmt.Fprintf(os.Stderr, "\nrules_go: Running 'bazel mod tidy' since %s changed...\n", strings.Join(diff, ", "))
+			err = runProcess([]string{"bazel", "mod", "tidy"}, os.Environ(), cwd)
```

Comment:
FYI: not everybody use the `bazel` binary from $PATH.

Some folks prefer using `bazelisk`, `b`, `bb`, `blaze`.

Could we make this configurable somehow? Or better, could we expose a starlark API for this?

Reply:
Good point, I'm now honoring the `BAZEL` environment variable.